pkg/installmanager/installmanager.go

```diff
@@ -351,6 +351,12 @@ func (m *InstallManager) Run() error {
 		m.log.WithError(err).Error("error while trying to preemptively clean up")
 		return err
 	}
+	// Fail the cluster provision if infraID is set. This can occur if previous install
+	// pod for current cluster provision is unexpectedly killed or evicted.
+	if provision.Spec.InfraID != nil {
+		m.log.WithError(err).Error("infraID is already set on the ClusterProvision. Unexpected install pod restart detected")
+		return fmt.Errorf("infraID is already set on the ClusterProvision. Unexpected install pod restart detected")
+	}
```

Comment:
This code will not update `ClusterProvisionFailed` condition with any relevant message about the failure. We have to check install pod logs to know the actual cause. The `ClusterProvisionFailed` gets updated by clusterprovision_controller with reason and message parsed from the install log. The install log is not stored until we run openshift-install binary. One option is we directly update ClusterProvisionFailed condition here. Any thoughts?

ClusterProvisionFailed condition with current code
```yaml
  - lastProbeTime: "2022-01-14T16:38:21Z"
    lastTransitionTime: "2022-01-14T16:38:21Z"
    message: Cluster install failed but installer log was not captured
    reason: UnknownError
    status: "True"
    type: ClusterProvisionFailed
```

Reply:
What would we update it to? If we don't have the install log, we don't have the install log.

Reply:
> What would we update it to? If we don't have the install log, we don't have the install log.

I was thinking of calling `setClusterProvisionCondtion` directly from here and set a `ClusterProvisionFailed` conditon along with an appropriate reason/message. Currently `clusterprovision` controller parses the install log to set this condition.

> If we did a `fmt.Println("HEY I FAILED BC INFRAID WAS NOT NIL")` would that be considered the "install log" for this provision attempt?
For this, we need to write stdout content to installer console log and then call a function to read the log and transfer it to clusterprovision. Currently, we start this process only after running the installer binary.

Reply:
I now see that the fmt.Println() calls are just so that this Pod's logs show the install output, but that isn't really considered the "install output".
Since we already are writing out to the ClusterProvision, I don't see why we couldn't also set this condition right here where we know a condition occurred. Just need to make sure the clusterprovision controller can gracefully handle the job itself updating those fields.

Reply:
No luck updating the cluster provision condition. The code [here](https://github.com/openshift/hive/blob/588f139c6122c62fd5232e7b47125bad103680fe/pkg/controller/clusterprovision/clusterprovision_controller.go#L383-L396) stomps over whatever condition I set. Since this is a rarely occurring situation we can get away with only an explicit message in the log I feel. Unless someone complains.